src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpExpressionCompiler.cs

```diff
@@ -4,16 +4,30 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.ExpressionEvaluator;
+using Microsoft.VisualStudio.Debugger;
+using Microsoft.VisualStudio.Debugger.CallStack;
 using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.ComponentInterfaces;
 using Microsoft.VisualStudio.Debugger.Evaluation;
 
 namespace Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator
 {
     [DkmReportNonFatalWatsonException(ExcludeExceptionType = typeof(NotImplementedException)), DkmContinueCorruptingException]
-    internal sealed class CSharpExpressionCompiler : ExpressionCompiler
+    internal sealed class CSharpExpressionCompiler : ExpressionCompiler, IDkmLanguageFrameDecoder, IDkmLanguageInstructionDecoder
```

Comment:
The same change will be needed for `VisualBasicExpressionCompiler`.

Reply:
Consider moving the interface implementations and the `FrameDecoder` and `LanguageInstructionDecoder` fields to the base class.

Reply:
Thank you!

---
In reply to: [144460725](https://github.com/dotnet/roslyn/pull/22681#discussion_r144460725) [](ancestors = 144460725)